vllm/v1/engine/__init__.py

```diff
@@ -82,7 +82,12 @@ class EngineCoreOutputs(
     # e.g. columnwise layout
 
     # [num_reqs]
-    outputs: List[EngineCoreOutput]
+    request_ids: List[str]
+    new_token_id_offsets: List[int]
```

Comment:
I didn't look closely at the corresponding logic but wondering if it might simplify things to make this a list of counts instead?

Also it may be worth making it optional, with `None` to mean the all counts are 1 (which will be a common case).

Reply:
I think the offsets make the code simpler.  I did make the list optional for the single token case though.